doc/conf.py

```diff
@@ -310,6 +310,10 @@
     "build/dts/api/compatibles/**/*",
 ]
 
+# -- Options for zephyr.domain --------------------------------------------
+
+zephyr_domain_apply_transforms = True
```

Comment:
until we settle on whether it's needed at all or needs to be tweaked, and before i forget, I would drop "_domain" from to be consistent with existing naming scheme of other Sphinx domains https://www.sphinx-doc.org/en/master/usage/configuration.html#options-for-the-c-domain 

```suggestion
zephyr_apply_transforms = True
```